Qute docs improvement

melloware opened the pull request:
Fix #44261

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -145,8 +145,8 @@ Hello Martin!
 - Organise your template files in the `/src/main/resources/templates` directory, by grouping them into one directory per resource class. So, if
-  your `ItemResource` class references two templates `hello` and `goodbye`, place them at `/src/main/resources/templates/ItemResource/hello.txt`
-  and `/src/main/resources/templates/ItemResource/goodbye.txt`. Grouping templates per resource class makes it easier to navigate to them.
+  your `HelloResource` class references two templates `hello` and `goodbye`, place them at `/src/main/resources/templates/ItemResource/hello.txt`
```
```suggestion
  your `HelloResource` class references two templates `hello` and `goodbye`, place them at `/src/main/resources/templates/HelloResource/hello.txt`
```

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -162,6 +162,13 @@ Hello {name}! <1>
 ----
 <1> `{name}` is a value expression that is evaluated when the template is rendered.
 
+.HelloResource/goodebye.txt
+[source]
+----
+Goodbye {name}! <1>
+----
+<1> `{name}` is a value expression that is evaluated when the template is rendered.
+
```
These lines should not be added, imho.
But it is important to rename [line 211](https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L211) from `HelloResource.java` to `GoodbyeResource.java`: https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L211-L226

neon-dev replied:
Also, add
```java
public static native TemplateInstance goodbye(String name);
```
after https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L194
<br>
And change https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L231 in GoodbyeResource.java to
```java
return HelloResource.Templates.goodbye(name);
```

neon-dev replied:
> These lines should not be added, imho.

Sorry, I was confused. They should stay of course.
But there's a typo in goodbye.txt: https://github.com/quarkusio/quarkus/blob/f32455e7570b08a41ace95fafa3f9223e6404104/docs/src/main/asciidoc/qute.adoc?plain=1#L165

And https://github.com/quarkusio/quarkus/blob/f32455e7570b08a41ace95fafa3f9223e6404104/docs/src/main/asciidoc/qute.adoc?plain=1#L156 should be changed to
```md
We'll start with two very simple templates:
```

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -201,7 +208,7 @@ NOTE: Once you have declared a `@CheckedTemplate` class, we will check that all
 Keep in mind this style of declaration allows you to reference templates declared in other resources too:
 
-.HelloResource.java
+.Goodbye.java
```
```suggestion
.GoodbyeResource.java
```

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -243,9 +250,11 @@ import io.quarkus.qute.CheckedTemplate;
 @CheckedTemplate
 public class Templates {
     public static native TemplateInstance hello(String name); <1>
+    public static native TemplateInstance goodbye(String name); <2>
```
Good catch, but this line needs also to be added in here:
https://github.com/quarkusio/quarkus/blob/f8e042f2caedeee44818d706980d4bc078e5c4e6/docs/src/main/asciidoc/qute.adoc?plain=1#L192-L195

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <github@undiscover.de>
Qute docs improvement
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <github@undiscover.de>
Qute docs improvement

mkouba reviewed (commented):
Hm, when I think about it, we probably should not make this change at all.

It does not make sense to introduce the `goodbye.txt` template in the example itself.

The sentence with `ItemResource` is about grouping templates per resource class (unrelated to the hello example).

CC @FroMage who is the author of the original text about type-safe templates.
CC @ia3andy

FroMage commented:
Yeah, it doesn't add a lot of value to document the second template, we could change the sentence to not mention the second template by name, though, so that people would not think that one is missing from the docs.

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -185,23 +192,25 @@ public class HelloResource {
 Keep in mind this style of declaration allows you to reference templates declared in other resources too:
 
-.HelloResource.java
+.GoodbyeResource.java
```
Also change
```java
@Path("goodbye")
public class GoodbyeResource {
```
to
```java
@Path("greeting")
public class GreetingResource {
```

neon-dev replied:
Don't forget to include [above changes](https://github.com/quarkusio/quarkus/pull/44289#discussion_r1850587033).

neon-dev commented on docs/src/main/asciidoc/qute.adoc:
```diff
@@ -162,6 +162,13 @@ Hello {name}! <1>
+----
+<1> `{name}` is a value expression that is evaluated when the template is rendered.
+
 Now let's declare and use those templates in the resource class.
```
```suggestion
Now let's declare and use this template in the resource class.
```

neon-dev reviewed (changes requested):
What do you think about these changes?

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Update docs/src/main/asciidoc/qute.adoc

Co-authored-by: Neon <1169307+neon-dev@users.noreply.github.com>

melloware committed:
Merge branch 'main' into Q44261

melloware commented:
I am gonna close this.  @neon-dev can you submit a new PR please?

neon-dev commented:
> I am gonna close this. @neon-dev can you submit a new PR please?

#44602
